### DISCUSSION TITLE
Tiffsave creates corrupted file with tile-width at 256

### DISCUSSION THREAD

**Discussion by fitnycdigitalinitiatives** [libvips/libvips#3458] [General]
## Questions, enhancements, tips, etc.

Please use libvips discussions for other topics.

https://github.com/libvips/libvips/discussions

## Bug report

**Describe the bug**
By default, we create pyramid tiff's for IIIF server purposes with tile-width and tile-height at 256. However, I noticed recently that this created a corrupted file for several images. However, when I tested that same image again with the default of 128, it was all good. My initial thought was that the source image was too small, but it seems fine with other images of comparable size, so not sure what's happening. Another aspect is the corrupted image itself still works with our IIIF server, but won't open in Photoshop (says it's corrupted) or mac preview.

**To Reproduce**
1. Use this [image](https://drive.google.com/file/d/16H5xz_r2VNIeHK9CFiF2WlavynEqaGEP/view?usp=sharing)
2. Use `vips tiffsave input output.tif --tile --pyramid --compression jpeg --tile-width 256 --tile-height 256` 

**Expected behavior**
Should create a proper pyramid tiff

**Actual behavior**
Creates corrupted tiff file

**Screenshot**

Corrupted file viewed through Preview

![image](https://user-images.githubusercontent.com/12830594/233659538-6f05c721-aad5-4753-b6aa-dcdee4deed75.png)

**Environment**
(please complete the following information)
 - OS: MacOS 10.15
 - Vips: 8.14.1

**Comment by jcupitt** [comment]
Hi @fitnycdigitalinitiatives,

I tried this:

```
$ vips tiffsave Arno_Penelope_Image\ 1.tif output.tif --tile --pyramid --compression jpeg --tile-width 256 --tile-height 256
```

And `output.tif` seems to work fine in all the programs I have (eog, gimp, tiffinfo, vd). I put a copy of my output file here:

http://www.rollthepotato.net/~john/output.tif

I noticed the file has a photoshop data packet. Perhaps pshop is complaining about a mismatch between the photoshop metadata and the file as saved by libvips?

I would try:

```
$ vips copy Arno_Penelope_Image\ 1.tif output.tif[tile,pyramid,compression=jpeg,tile-width=256,tile-height=256,strip]
```

ie. add `strip` in order to make an output tiff with no extra metadata. If that works, then I guess you need to remove the `photoshop-data` metadata item before saving as tiff.

**Reply by jcupitt** [reply]
... macOS preview only supports very basic TIFF image types. I think I would not expect it to work for tiled jpeg TIFFs.

**Comment by fitnycdigitalinitiatives** [comment]
Hi John,

Thanks for the response.

I ran it again with strip, but still the same problems. Again, what's curious is that if I change the tile height/width to 128, everything works fine, with or without the metadata. As for Preview, I've never had trouble opening up the p-tiffs we're creating in it, or with Photoshop. I'm not terribly worried because it's all still working good with our image server, but it makes me wonder what's going on.

-Joe

**Reply by jcupitt** [reply]
Huh that's weird. Then I don't know what the problem is, sorry.

Does that `output.tif` I posted work in photoshop?

**Reply by fitnycdigitalinitiatives** [reply]
No, the file you shared has the same result in photoshop and preview:

<img width="1701" alt="image" src="https://user-images.githubusercontent.com/12830594/233678733-4082dd13-7926-48aa-9ddc-52702b278851.png">

Well, thanks for taking a look.
